### PR TITLE
Add in device support

### PR DESCRIPTION
**Pull request by gregorygtseng**

### CODE REVIEW

**Comment by bradrydzewski** [comment]
really great. only minor comment would be to add `d.Devices = nil` to the following function as well:
https://github.com/gregorygtseng/drone-exec/blob/device-support/parser/funcs.go#L152

but otherwise, this is a great PR. Thanks so much!

**Comment by gregorygtseng** [comment]
Hmm.. Should we be clearing those options if it's escalated?

**Comment by bradrydzewski** [comment]
> Hmm.. Should we be clearing those options if it's escalated?

yes, these are special plugins that run dind and wouldn't be able to access this parameter anyway. It is just an extra safety precaution.

**Comment by gregorygtseng** [comment]
thanks for explaining the context!

**Comment by bradrydzewski** [comment]
LGTM

File: runner/utils.go [hunk 1]
```diff
@@ -52,6 +52,20 @@ func toContainerConfig(n *parser.DockerNode) *dockerclient.ContainerConfig {
 		config.HostConfig.Binds = append(config.HostConfig.Binds, path)
 	}
 
+	for _, path := range n.Devices {
+		if strings.Index(path, ":") == -1 {
+			// Invalid config will just silently fail...
+			continue
+		}
+		parts := strings.Split(path, ":")
```
Comment (tboerger) [inline runner/utils.go]: Shouldn't that be SplitN?

Comment (gregorygtseng) [inline runner/utils.go]: Why SplitN?

Comment (tboerger) [inline runner/utils.go]: You always expect 2 parts.

**Pull request merged**